src/zjs_buffer.c

```diff
@@ -280,20 +280,17 @@ static jerry_value_t zjs_buffer_write_string(const jerry_value_t function_obj_va
 
     // Check if the encoding string is anything other than utf8
     if (argc > 3) {
-        jerry_value_t arg4 = argv[3];
-        jerry_size_t arg4_sz = jerry_get_string_size(arg4);
-
-        if (arg4_sz > 4096) {
-            return zjs_error("zjs_buffer_write_string: encoding arg string is too long");
+        // ask for one more char than needed to make sure not just prefix match
+        jerry_size_t size = 5;
```

Comment:
Since the string we're checking it against is "utf8", we only need the first five bytes to compare. This could be made more clear w/ comments or perhaps a standard compare function.